File: Finance.Domain/Entities/UserSession/ImportSession.cs
using Finance.Domain.Entities.Core;

namespace Finance.Domain.Entities.UserSession
{
    public class ImportSession
    {
        public Guid Id { get; set; }

        public Guid UserId { get; set; }
        public User User { get; set; } = null!;

        public string? FileName { get; set; }

        public Guid? SourceAccountId { get; set; }
        public Account? SourceAccount { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public string Label { get; set; } = "user_imports";

        public bool IsDeletable { get; set; } = true;

        public string Strategy { get; set; } = "bayesian_statistics";

        public bool HasExclusions { get; set; }

        public string ColumnMappingsJson { get; set; } = "{}";

        public bool IsArchived { get; set; }

        public string Status { get; set; } = "Active";

        public ICollection<ImportSessionRow> Rows { get; set; } = new List<ImportSessionRow>();
    }
}

File: Finance.WebAPI/finance-spa/models/import-session.ts
export type ImportSessionRow = {
  id: string
  rowIndex: number
  values: string[]
  destinationAccountId: string | null
  destinationAccountError: string | null
  mappingSource: string
  learningConfidenceScore: number | null
  addedToLedgerAt: string | null
  postedTransactionId: string | null
  projectBalance: number
}

export type ImportSession = {
  id: string
  createdAt: string
  createdByUserId: string
  createdByUserName: string
  fileName: string | null
  sourceAccountId: string | null
  sourceAccountName: string | null
  label: "user_imports" | "user_import_chunked" | "account_deletion_sessions" | string
  isDeletable: boolean
  strategy: string
  isArchived: boolean
  status: string
  columnMappings: Record<number, string>
  rows: ImportSessionRow[]
}

export type CreateImportSessionRowInput = {
  rowIndex: number
  values: string[]
  destinationAccountId: string | null
  destinationAccountError: string | null
  addedToLedgerAt?: string | null
  postedTransactionId?: string | null
}

export type CreateImportSessionInput = {
  fileName: string | null
  sourceAccountId: string | null
  label?: "user_imports" | "user_import_chunked" | "account_deletion_sessions" | string
  strategy?: string
  isArchived?: boolean
  status?: string
  columnMappings: Record<number, string>
  rows: CreateImportSessionRowInput[]
}
